### PR TITLE
bootstrap: add featuregate CR

### PR DESCRIPTION
**Pull request by deads2k**
Config resources need to exist to kick off many operators.  Most of these should never be misssing.
FeatureGate is one such example that we should always have and we prevent its deletion.

### CODE REVIEW

**Comment by deads2k** [comment]
/hold

I think I found past-me's pull making this kind of work from cluster-config-operator

**Comment by deads2k** [comment]
https://github.com/openshift/cluster-config-operator/pull/15 may get the job done for initial creation

File: pkg/asset/manifests/featuregate.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+// Name returns a human friendly name for the asset.
+func (*FeatureGate) Name() string {
+	return "FeatureGate Config"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate the asset.
+func (*FeatureGate) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
```
Comment (staebler) [inline pkg/asset/manifests/featuregate.go]: The InstallConfig dependency is unnecessary.

Comment (deads2k) [reply pkg/asset/manifests/featuregate.go]: > The InstallConfig dependency is unnecessary.

removing it appears to have caused multiple failures.

Comment (staebler) [reply pkg/asset/manifests/featuregate.go]: I should have been more complete with my comment. You need to also remove the code in `Generate` where the InstallConfig is retrieved.

https://github.com/openshift/installer/blob/5b62a4df18af00285191b512b3965fd55fb3611c/pkg/asset/manifests/featuregate.go#L38-L39

Comment (deads2k) [reply pkg/asset/manifests/featuregate.go]: > I should have been more complete with my comment. You need to also remove the code in `Generate` where the InstallConfig is retrieved.

and now I feel slow.

**Comment by eparis** [comment]
https://github.com/openshift/api/pull/278

**Comment by deads2k** [comment]
alright, given the stuck MCD issue, I'm going down a path of modifying the CVO to handle our use-case

**Pull request closed without merging**